lib/_stream_writable.js

```diff
@@ -409,14 +400,13 @@ ObjectDefineProperty(Writable.prototype, 'writableCorked', {
 // If we're already writing something, then just put this
 // in the queue, and wait our turn.  Otherwise, call _write
 // If we return false, then we need a drain event, so set that flag.
-function writeOrBuffer(stream, state, isBuf, chunk, encoding, cb) {
-  if (!isBuf) {
-    var newChunk = decodeChunk(state, chunk, encoding);
-    if (chunk !== newChunk) {
-      isBuf = true;
-      encoding = 'buffer';
-      chunk = newChunk;
-    }
+function writeOrBuffer(stream, state, chunk, encoding, cb) {
+  if (!state.objectMode &&
+      state.decodeStrings !== false &&
+      encoding !== 'buffer' &&
+      typeof chunk === 'string') {
```

Comment:
Nit: having `isBuf` in this function would allow us to simplify this if to `if (!isBuf && state.decodeStrings !== false && typeof chunk === 'string')` I think.

Reply:
In my opinion it is simple to just check encoding. There does not seem to be any perf diff.